packages/firebase_crashlytics/lib/src/firebase_crashlytics.dart

```diff
@@ -0,0 +1,213 @@
+    // https://github.com/flutter/flutter/issues/26431
+    // ignore: strong_mode_implicit_dynamic_method
+    final bool result = await channel.invokeMethod('Crashlytics#isDebuggable');
+    return result;
+  }
+
+  Future<String> getVersion() async {
+    // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
+    // https://github.com/flutter/flutter/issues/26431
+    // ignore: strong_mode_implicit_dynamic_method
+    final String result = await channel.invokeMethod('Crashlytics#getVersion');
+    return result;
+  }
+
+  /// Add text logging that will be sent with your next report. `msg` will be
+  /// printed to the console when in debug mode.
+  void log(String msg) {
+    _logSize += Uint8List.fromList(msg.codeUnits).length;
+    logs.add(msg);
+    // Remove oldest log till logSize is no more than 64K.
+    while (_logSize > 65536) {
+      final String first = logs.removeFirst();
+      _logSize -= Uint8List.fromList(first.codeUnits).length;
+    }
+  }
+
+  void setKey(String key, dynamic value) {
+    // Check that only 64 keys are set.
+    if (keys.containsKey(key) || keys.length <= 64) {
+      keys[key] = value;
+    }
+  }
+
+  void setBool(String key, bool value) {
+    setKey(key, value);
+  }
+
+  void setDouble(String key, double value) {
+    setKey(key, value);
+  }
+
+  void setInt(String key, int value) {
+    setKey(key, value);
+  }
+
+  void setString(String key, String value) {
+    setKey(key, value);
+  }
+
+  Future<void> setUserEmail(String email) async {
+    // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
+    // https://github.com/flutter/flutter/issues/26431
+    // ignore: strong_mode_implicit_dynamic_method
+    await channel.invokeMethod(
+        'Crashlytics#setUserEmail', <String, dynamic>{'email': email});
+  }
+
+  Future<void> setUserIdentifier(String identifier) async {
+    // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
+    // https://github.com/flutter/flutter/issues/26431
+    // ignore: strong_mode_implicit_dynamic_method
+    await channel.invokeMethod('Crashlytics#setUserIdentifier',
+        <String, dynamic>{'identifier': identifier});
+  }
+
+  Future<void> setUserName(String name) async {
+    // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
+    // https://github.com/flutter/flutter/issues/26431
+    // ignore: strong_mode_implicit_dynamic_method
+    await channel.invokeMethod(
+        'Crashlytics#setUserName', <String, dynamic>{'name': name});
+  }
+
+  @visibleForTesting
+  Future<void> sendLogs() async {
+    for (int i = 0; i < logs.length; i++) {
+      // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
+      // https://github.com/flutter/flutter/issues/26431
+      // ignore: strong_mode_implicit_dynamic_method
+      await channel.invokeMethod('Crashlytics#log', <String, dynamic>{
```

Comment:
This seems like a lot of unnecessary async waiting, especially if you have big logs. Can we send everything to the native side as an array argument to onError and log it there?

Reply:
Done

Reply:
Removed since logs are sent as part of error reporting.